cmd/ceremony/main.go

```diff
@@ -476,6 +546,75 @@ func ocspRespCeremony(configBytes []byte) error {
 	return nil
 }
 
+func crlCeremony(configBytes []byte) error {
+	var config crlConfig
+	err := yaml.UnmarshalStrict(configBytes, &config)
+	if err != nil {
+		return fmt.Errorf("failed to parse config: %s", err)
+	}
+	if err := config.validate(); err != nil {
+		return fmt.Errorf("failed to validate config: %s", err)
+	}
+
+	ctx, session, err := pkcs11helpers.Initialize(config.PKCS11.Module, config.PKCS11.SigningSlot, config.PKCS11.PIN)
+	if err != nil {
+		return fmt.Errorf("failed to setup session and PKCS#11 context for slot %d: %s", config.PKCS11.SigningSlot, err)
+	}
+	log.Printf("Opened PKCS#11 session for slot %d\n", config.PKCS11.SigningSlot)
+	keyID, err := hex.DecodeString(config.PKCS11.SigningKeyID)
+	if err != nil {
+		return fmt.Errorf("failed to decode key-id: %s", err)
+	}
+	signer, err := newSigner(ctx, session, config.PKCS11.SigningLabel, keyID)
+	if err != nil {
+		return fmt.Errorf("failed to retrieve private key handle: %s", err)
+	}
+	log.Println("Retrieved private key handle")
+
+	issuer, err := core.LoadCert(config.Inputs.IssuerCertificatePath)
+	if err != nil {
+		return fmt.Errorf("failed to load issuer certificate %q: %s", config.Inputs.IssuerCertificatePath, err)
+	}
+
+	dateLayout := "2006-01-02 15:04:05"
+
+	thisUpdate, err := time.Parse(dateLayout, config.CRLProfile.ThisUpdate)
+	if err != nil {
+		return fmt.Errorf("unable to parse crl-profile.this-update: %s", err)
+	}
+	nextUpdate, err := time.Parse(dateLayout, config.CRLProfile.NextUpdate)
+	if err != nil {
+		return fmt.Errorf("unable to parse crl-profile.next-update: %s", err)
+	}
+
+	var revokedCertificates []pkix.RevokedCertificate
+	for _, rc := range config.CRLProfile.RevokedCertificates {
+		cert, err := core.LoadCert(rc.CertificatePath)
+		if err != nil {
+			return fmt.Errorf("failed to load revoked certificate %q: %s", rc.CertificatePath, err)
+		}
+		revokedAt, err := time.Parse(dateLayout, rc.RevocationDate)
+		if err != nil {
+			return fmt.Errorf("unable to parse crl-profile.revoked-certificates.revocation-date")
+		}
+		revokedCertificates = append(revokedCertificates, pkix.RevokedCertificate{
+			SerialNumber:   cert.SerialNumber,
+			RevocationTime: revokedAt,
+		})
+	}
+
+	crlBytes, err := generateCRL(signer, issuer, thisUpdate, nextUpdate, config.CRLProfile.Number, revokedCertificates)
+	if err != nil {
+		return err
+	}
+
+	if err := ioutil.WriteFile(config.Outputs.CRLPath, crlBytes, 0644); err != nil {
```

Comment:
In the certificate case, we print the certificate PEM to stdout after signing and before writing to a file, so in case there is some problem writing to an output file at least we can in theory copy down whatever's on the screen. I think we should do that here too (and, perhaps separately, for the OCSP ceremony case).